contracts/InterchainTokenService.sol

```diff
@@ -1101,6 +1101,7 @@ contract InterchainTokenService is
 
         address minter;
         if (bytes(minterBytes).length != 0) minter = minterBytes.toAddress();
+        if (minter == address(this)) revert InvalidMinter(minter);
```

Comment:
this helper is used when processing received deploy call as well, where it doesn't make sense to revert since it'll be a no-op anyways. Having thought more about this, I think it's fine to not check in ITS. And just keep the one in the factory. ITS deploy is flexible so the user can set minter to whatever anyways, so we only really care about trustless factory deployments.

Reply:
Got it. updated to keep the checking only in the factory. also updated docstring in the factory about the minter.